Module4_Labs/Lab1_BFS_Maze/Cards/3.md

```diff
@@ -4,8 +4,23 @@
 
 <!--badges={Python:50,CreativeThinker:50}-->
 
-`getAdjacentSpaces()` should get all neighbors of a space in a given maze. It should return a list of neighbors that have not been visited. Neighbors in this case mean the possible directions one can go in the maze.
+`getAdjacentSpaces()` should get all neighbors of a space in a given maze. It should return a list of adjacent neighbors (tuples of spaces) that have not been visited. Valid neighbors in this case mean the possible directions one can go in the maze (meaning that they shouldn't be walls and should not have been visited yet).  
 
-* Make sure to check if the space is actually accessible and not a wall!
+** Parameters of `getAjacentSpaces()` **
+
+The function will take in 3 parameters which are:
+
+* `maze` - This is the 2D array maze
+
+* `space` - This is a `tuple` which will contain the coordinates of the current space that we are searching on, we want to find the available adjacent spaces around this coordinate.
+
+* `visited` - This is a `list` of all the spaces we have visited. We will use this `list` to check whether a neighbor can be found `in` this list, if so then it means that the neighbor has already been visited and thus it will not be a valid neighbor. We need to check whether a neighbor has been visited or not because we do not want to process again the spaces which we have visited. Revisitting visited spaces can create an endless cycle of searching. 
+
+
+Our function should simply try to look at the surrounding spaces and see whether the spaces around it is a valid neighbor. We could do our search in a well defined order of clockwise direction, starting from looking at the neighbor in the North , then East, then South, and finally West. 
```

Comment:
add a note that this is important so that their output matches when submitting their code.